src/Model/Group.php

```diff
@@ -300,20 +317,51 @@ public static function removeMemberByName($uid, $name, $cid)
 	/**
 	 * @brief Returns the combined list of contact ids from a group id list
 	 *
+	 * @param int     $uid
 	 * @param array   $group_ids
 	 * @param boolean $check_dead
 	 * @return array
 	 * @throws \Exception
 	 */
-	public static function expand($group_ids, $check_dead = false)
+	public static function expand($uid, array $group_ids, $check_dead = false)
 	{
 		if (!is_array($group_ids) || !count($group_ids)) {
 			return [];
 		}
 
-		$stmt = DBA::select('group_member', ['contact-id'], ['gid' => $group_ids]);
-
 		$return = [];
+
+		$key = array_search(self::FOLLOWERS, $group_ids);
+		if ($key !== false) {
+			$followersStmt = Contact::select(['id'], [
+				'uid' => $uid,
+				'rel' => [Contact::FOLLOWER, Contact::FRIEND],
+				'protocol' => Protocol::SUPPORT_PRIVATE,
+			]);
+
+			while($follower = DBA::fetch($followersStmt)) {
+				$return[] = $follower['id'];
+			}
+
+			unset($group_ids[$key]);
+		}
+
+		$key = array_search(self::MUTUALS, $group_ids);
+		if ($key !== false) {
+			$mutualsStmt = Contact::select(['id'], [
+				'uid' => $uid,
+				'rel' => [Contact::FRIEND],
+				'protocol' => Protocol::SUPPORT_PRIVATE,
+			]);
+
+			while($mutual = DBA::fetch($mutualsStmt)) {
+				$return[] = $mutual['id'];
+			}
+
+			unset($group_ids[$key]);
+		}
+
+		$stmt = DBA::select('group_member', ['contact-id'], ['gid' => $group_ids]);
```

Comment:
I just realized that none of the ```DBA::select``` functions in this file is calling ```DBA::close``` right now. I think it would be cleaner to add this.